src/goLanguageServer.ts

```diff
@@ -589,3 +572,53 @@ async function goplsVersion(goplsPath: string): Promise<string> {
 	//
 	return split[1];
 }
+
+async function goProxyRequest(tool: Tool, endpoint: string): Promise<any> {
+	const proxies = await goProxy();
+	if (!proxies) {
+		return null;
+	}
+	// Try each URL set in the user's GOPROXY environment variable.
+	// If none is set, don't make the request.
+	for (const proxy of proxies) {
+		if (proxy === 'direct') {
+			continue;
+		}
+		const url = `${proxy}/${tool.importPath}/@v/${endpoint}`;
+		let data: string;
+		try {
+			data = await WebRequest.json<string>(url, {
+				throwResponseError: true,
+			});
+		} catch (e) {
+			return null;
+		}
+		return data;
+	}
+	return null;
+}
+
+async function goProxy(): Promise<string[]>  {
```

Comment:
We plan to do this not more than once per each VS Code session right?
If so, then [on start up](https://github.com/microsoft/vscode-go/blob/0.11.7/src/goInstallTools.ts#L330), we do run `go env GOPATH GOROOT`.
We can append GOPROXY to that and have the value stored in `process.env['GOPROXY']` and skip the below.

Reply:
I believe re-using that `go env` call is a good move. If you agree, you can cherry pick [this commit of mine](https://github.com/ramya-rao-a/vscode-go/commit/1d7700a1ef30a9b1c6fd5f15cccfbb921103267c)

Reply:
Done, thanks!